specification/keyvault/Security.KeyVault.BackupRestore/routes.tsp

```diff
@@ -0,0 +1,256 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "./models.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ClientGenerator.Core;
+
+namespace KeyVault;
+
+/**
+ * Returns the status of full backup operation
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+#suppress "@azure-tools/typespec-azure-core/use-standard-names" "Operation name is already established"
+@route("/backup/{jobId}/pending")
+@get
+op fullBackupStatus is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The id returned as part of the backup request
+     */
+    @path
+    jobId: string;
+  },
+  FullBackupOperation,
+  {},
+  KeyVaultError
+>;
+
+/**
+ * Creates a full backup using a user-provided SAS token to an Azure blob storage container.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@pollingOperation(fullBackupStatus)
+@finalOperation(fullBackupStatus)
+@useFinalStateVia("azure-async-operation")
+@route("/backup")
+@post
+op fullBackup is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * Azure blob shared access signature token pointing to a valid Azure blob container where full backup needs to be stored. This token needs to be valid for at least next 24 hours from the time of making this call.
+     */
+    @body azureStorageBlobContainerUri?: SASTokenParameter;
+  },
+  {
+    @statusCode _: 202;
+
+    /**
+     * The URI to poll for completion status.
+     */
+    @pollingLocation
+    @header("Azure-AsyncOperation")
+    azureAsyncOperation?: string;
+
+    /**
+     * The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.
+     */
+    @header("Retry-After") retryAfter?: int32;
+
+    /**
+     * The backup operation.
+     */
+    @body response: FullBackupOperation;
```

Comment:
Does the service actually return/includ this body for the 202 response? Or is this inheriting the lies from the swagger/LRO pattern that we used to keep auto rest (classic) happy?

Reply:
This is accurate: https://github.com/Azure/azure-sdk-assets/blob/python/keyvault/azure-keyvault-administration_b16e831fc9/python/sdk/keyvault/azure-keyvault-administration/tests/recordings/test_backup_client.pyTestBackupClientTeststest_full_backup_and_restore%5B7.5%5D.json#L62

Reply:
Ok. But. Follow-up question, Is the `null` value in the recording accurate? If so, the typespec attribute should be nullable as well.

Reply:
Good point -- the `Error` itself [should be nullable](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/common.json#L71), so I [updated](https://github.com/Azure/azure-rest-api-specs/pull/30239/commits/c30f473a0907bd9868c8d28ae9fcceaf1c233d78) the Common definition. Aside from `endTime` and `error` though, the [original spec](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.5/backuprestore.json#L366) doesn't mark any other attributes as nullable.